modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/VersionChain.java

```diff
@@ -131,7 +131,29 @@ public boolean hasCommittedVersions() {
         return newestCommittedLink() != NULL_LINK;
     }
 
-    /** {@inheritDoc} */
+    /**
+     * Returns {@code true} if there is a link to the next version.
+     */
+    public boolean hasNextLink() {
+        return nextLink != NULL_LINK;
+    }
+
+    /**
+     * Returns {@code true} if there is a link to the head version.
+     */
+    public boolean hasHeadLink() {
+        return headLink != NULL_LINK;
+    }
+
+    /**
+     * Creates a copy of the version chain with new next link.
+     *
+     * @param nextLink New next link.
+     */
+    public VersionChain setNextLink(long nextLink) {
```

Comment:
Maybe you should rename it to "withNextLink", "set" is usually mutating given object instead of creating a new one

Reply:
Fix it